sdk/core/corehttp/corehttp/streaming/_stream.py

```diff
@@ -0,0 +1,178 @@
+# --------------------------------------------------------------------------
+#
+# Copyright (c) Microsoft Corporation. All rights reserved.
+#
+# The MIT License (MIT)
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the ""Software""), to
+# deal in the Software without restriction, including without limitation the
+# rights to use, copy, modify, merge, publish, distribute, sublicense, and/or
+# sell copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in
+# all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED *AS IS*, WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+# FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
+# IN THE SOFTWARE.
+#
+# --------------------------------------------------------------------------
+
+from types import TracebackType
+from typing import Iterator, AsyncIterator, TypeVar, Callable, Any, Optional, Type
+
+from typing_extensions import Self
+
+from ..rest import HttpRequest, HttpResponse, AsyncHttpResponse
+from ..runtime.pipeline import PipelineResponse
+from ._decoders import JSONLDecoder, AsyncJSONLDecoder, StreamDecoder, AsyncStreamDecoder
+
+
+ReturnType = TypeVar("ReturnType")
+
+
+class Stream(Iterator[ReturnType]):
+    """Stream class for streaming JSONL or Server-Sent Events (SSE).
+
+    :keyword response: The pipeline response object.
+    :paramtype response: ~corehttp.runtime.pipeline.PipelineResponse
+    :keyword deserialization_callback: A callback that takes HttpResponse and JSON and
```

Comment:
Looks like a `PipelineResponse` is actually being sent to the callback rather than an `HttpResponse`?

Reply:
Oops, good catch!